### DISCUSSION TITLE
Airflow dag run prioritization

### DISCUSSION THREAD

**Discussion by boublilm** [apache/airflow#32134] [Q&A]
Hello all, I've come across an issue with Airflow 2.6.2 after upgrading from 2.5.1 that I'd like to discuss. Consider a scenario where we have 3 scheduled DAG runs, and the dag is configured to have 1 active run. If the most recent DAG run is currently executing and the remaining two are simply scheduled, attempting to clear the active DAG run task doesn't shift priority to the earlier scheduled runs as expected. Instead, the system seems to re-run the most recent (cleared) DAG task again.

This is contrary to the behavior we observed in the previous version where clearing a task would trigger a shift in priority towards older DAG runs. Currently, it seems like this prioritization of older DAG runs only occurs if the DAG run is in a failed or success state.

I have searched the changelogs but found no mention of changes to the prioritization configuration. Could there have been an update affecting the prioritization of DAG runs?

**Comment by vadipo** [comment]
I think that the behavior that changed is that in the current version clearing a task does not change the dagrun state back to "scheduled" - (that change in state allowed the scheduler to prioritize older runs as it should **by design**) -
it is look like unintentional bad side-effect to some kind of optimization .
I suspect this change: https://github.com/apache/airflow/pull/30125.
@dimonchik-suvorov what do you think?
